c/src/ml-api-service-remote.c

```diff
@@ -400,17 +407,25 @@ _mlrs_process_remote_service (nns_edge_data_h data_h)
             "Failed to get data from uri: %s.", (gchar *) data);
       }
       ret = ml_service_set_pipeline (service_key, (gchar *) array->data);
+
       g_byte_array_free (array, TRUE);
+      event_type = ML_SERVICE_EVENT_PIPELINE_REGISTERED;
       break;
     }
     case ML_REMOTE_SERVICE_TYPE_PIPELINE_RAW:
       ret = ml_service_set_pipeline (service_key, (gchar *) data);
+      event_type = ML_SERVICE_EVENT_PIPELINE_REGISTERED;
       break;
     default:
       _ml_error_report ("Unknown service type or not supported yet. "
           "Service num: %d", service_type);
       break;
   }
+
+  if (remote_s->event_cb) {
```

Comment:
plz consider adding condition before invoking the callback, to prevent unnecessary event.
```
if (remote_s && event_type != NNS_EDGE_EVENT_TYPE_UNKNOWN) {
  if (remote_s->event_cb)
    remote_s->event_cb (event_type, remote_s->user_data);
}
```